src/languages/vbnet.js

```diff
@@ -1,63 +1,114 @@
-/*
-Language: Visual Basic .NET
-Description: Visual Basic .NET (VB.NET) is a multi-paradigm, object-oriented programming language, implemented on the .NET Framework.
-Author: Poren Chiang <ren.chiang@gmail.com>
-Website: https://docs.microsoft.com/en-us/dotnet/visual-basic/getting-started/
-*/
+/**
+ * Language: Visual Basic .NET
+ * Description: Visual Basic .NET (VB.NET) is a multi-paradigm, object-oriented programming language, implemented on the .NET Framework.
+ * Authors: Poren Chiang <ren.chiang@gmail.com>, Jan Pilzer
+ * Website: https://docs.microsoft.com/dotnet/visual-basic/getting-started/
+ */
 
 export default function(hljs) {
+  const CHAR = {
+    className: 'string',
+    begin: /"(""|[^/n])"C\b/,
+  };
+
+  const STRING = {
+    className: 'string',
+    begin: /"/,
+    end: /"/,
+    illegal: /\n/,
+    contains: [{ begin: /""/ }], // double quote escape
+  };
+
+  /**
+   * Date Literals consist of a date or a time or both separated by whitespace, surrounded by #.
+   * Date can be in M/D/YYYY or YYYY-MM-DD format.
+   * A time value may be specified either using a 24-hour value (H:mm[:ss]) or a 12-hour value (h[:mm[:ss]] A)
+   */
+  const DATE = {
+    className: 'literal',
+    begin: /# *(((\d+[-/]){2}\d+)|((\d+(:\d+){1,2} ?(AM|PM)?)|(\d+ ?(AM|PM)))|((\d+[-/]){2}\d+ ((\d+(:\d+){1,2} ?(AM|PM)?)|(\d+ ?(AM|PM))))) *#/,
```

Comment:
[Maintainability] Ditto on `variants` here... if there are a few different possibilities (it looks like it) use variants or `regex.either`...  I'd expect to at least see two variants:

m/d/y vs y/m/d

Reply:
Much better though!

Reply:
There are two date formats (m/d/y and y-m-d) and two time formats (12h and 24h).
They are both optional (d, t, or dt) for 8 total combinations.
Any preference for listing variants?

Reply:
4 is probably fine... just add some visiblity

- 12h
- 24h
- YMD time (you can use `regex.concat` and `regex.either` to make this nicer
- DMY time (you can use `regex.concat` and `regex.either` to make this nicer